src/policy/immix/immixspace.rs

```diff
@@ -278,12 +280,13 @@ impl<VM: VMBinding> ImmixSpace<VM> {
         args: crate::policy::space::PlanCreateSpaceArgs<VM>,
         space_args: ImmixSpaceArgs,
     ) -> Self {
-        #[cfg(feature = "immix_non_moving")]
-        info!(
-            "Creating non-moving ImmixSpace: {}. Block size: 2^{}",
-            args.name,
-            Block::LOG_BYTES
-        );
+        if space_args.never_move_objects {
```

Comment:
We can make `space_args` mutable

```diff
-space_args: ImmixSpaceArgs,
+mut space_args: ImmixSpaceArgs,
```

and override `never_move_objects` here if `cfg!(feature = "immix_non_moving")` is true.  We can also print a log here, telling the user that this is overridden.